.github/workflows/updatecli.yml

```diff
@@ -0,0 +1,36 @@
+name: "Updatecli: Dependency Management"
+
+on:
+  schedule:
+    # Runs at 06 PM UTC
+    - cron: '0 18 * * *'
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+permissions:
+  contents: write
+  issues: write
+  pull-requests: write
+
+jobs:
+  updatecli:
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/main'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Install Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: '1.20.1'
```

Comment:
As I commented below in the UpdateCLI file, this will not work out of the box. To avoid such updates, we could use `stable` instead of the specific version. WDYT?

Reply:
yeah, I think setting this to stable would resolve the need to keep it up to date.

For future me, here is a link to the documentation about that: https://github.com/actions/setup-go#using-stableoldstable-aliases